server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -718,6 +718,74 @@ public Map<String, UploadedSegmentMetadata> getSegmentsUploadedToRemoteStore() {
         return Collections.unmodifiableMap(this.segmentsUploadedToRemoteStore);
     }
 
+    private Map<String, String> populatePreviousMetadata(
```

Comment:
Lets write exhaustive UTs for this functions. 

 I think we can simplify the code finding previous and next metadata , by using  a binary tree . All the un-deleted metadata files can be inserted into a binary tree . Then prev and next metadata would just be finding nearest neighbours on both side . However the creation of tree itself would be n log(n), whereas the current solution is more optimal o(n) .

Reply:
The current code looks unwieldy to me . Is there a way we can simplify this a bit ? I will also give a thought about the same and see if i have some ideas .

Reply:
updated the code now.